### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by DavisRayM**
- Switch from using `ANXS.python` to [`DavisRayM.python`](https://github.com/DavisRayM/python)
- Fix an issue where the git key file would be treated as the git key content
- Add a task that optionally installs packages from github using pip

## Reviewer Note

Decided to diverge from the ANXS.python role since it has a few weird tasks I wasn't too sure how to handle on the role.

1. The role uses `/usr/bin/python` to install python. _This is not configurable. Specifying the `python_source_version` doesn't seem to change anything... As far as I could tell._ In most cases if the server has both python2 and python3 the role will install pip for python2 unless `/usr/bin/python` is linked to python3.

2. It doesn't specify the executable for pip tasks to the `python_pip_location` variable it has. This is usually fine but in servers where python3 is present ansible defaults to using pip3 which might not be installed and thus this task fails. This can be fixed by changing the `ansible_python_interpreter` to use `/usr/bin/python` but then it just installs pip for python2 and the packages for python2 also....

_TL;DR main issue with the role is that it doesn't install python3 and I don't believe it's possible to configure it to do so as it is_

I think it'll be better to maintain our own fork of the ansible role as it doesn't seem to be actively maintained(_Last update was in November 2018_). Ideally we wouldn't use the repo in my account but move it to the `onaio` organization if possible.

### CODE REVIEW

**Comment by moshthepitt** [comment]
> Switch from using ANXS.python to DavisRayM.python

Is there a danger of this fork diverging further from `ANXS.python`?  If yes, what's the plan for dealing with that?

File: meta/main.yml [hunk 1]
```diff
@@ -10,9 +10,11 @@ galaxy_info:
       - all
 
 dependencies:
- - role: ANXS.python
+ - role: DavisRayM.python
```
Comment (ukanga) [inline meta/main.yml:13]: Why do we need to switch?

File: tasks/install.yml [hunk 2]
```diff
@@ -11,25 +11,6 @@
...
-    update_cache: "yes"
-
-- name: Install system-wide dependencies
-  apt:
-    name: "{{ django_system_wide_dependencies }}"
-    state: latest
-    update_cache: yes
-    cache_valid_time: 600
-
 - name: Delete virtualenv
```
Comment (ukanga) [inline tasks/install.yml:14]: Has this been moved or it is no longer necessary?

Comment (ukanga) [reply tasks/install.yml:14]: There might be Django dependencies that may not be applicable to python in general, for example installing zlib1g-dev which may be a dependency on your Django application for the requirements to be successfully installed.

**Comment by DavisRayM** [comment]
> > Switch from using ANXS.python to DavisRayM.python
> 
> Is there a danger of this fork diverging further from `ANXS.python`? If yes, what's the plan for dealing with that?

The ANXS.python package is no longer being supported, I don't believe there is any danger in diverging from it..

File: meta/main.yml (hunk 1 above)
Comment (DavisRayM) [reply meta/main.yml:13]: The `ANXS.python` ansible role doesn't use python3 to install pip (Doesn't install python3 too). It uses whatever `/usr/bin/python` links to which is usually python2 and this causes a few issues....

The `ANXS.python` role also doesn't specify the pip executable to use(defaulting to pip3 if python3 is present on the server) but the role installed pip2 which ends up failing unless you switch `ansible_python_interpreter` to `/usr/bin/python3` when using this playbook on a server with python3 installed

File: tasks/install.yml (hunk 2 above)
Comment (DavisRayM) [reply tasks/install.yml:14]: This was moved. Mhh I see, that make sense didn't think about it that way. I'll revert this back.

Comment (DavisRayM) [reply tasks/install.yml]: Reverted the `Install system-wide dependencies`. I've still left out the `Add Python PPA` tasks since I think it's better to have within the role that installs python.... thoughts ? @ukanga

**Comment by DavisRayM** [comment]
Found this interesting ansible role that utilizes `pythonz` to install a specific python version I think this might also be a good role to use https://github.com/pyslackers/ansible-role-python. I haven't used the `pythonz` tools before so I'm not too sure how it installs and links python exactly.

I'll investigate this a bit.

File: meta/main.yml [hunk 3]
```diff
@@ -10,9 +10,13 @@ galaxy_info:
...
    become: true
    become_user: "root"
    python_source_version: "{{ django_python_source_version }}"
+   python_packages:
```
Comment (moshthepitt) [inline meta/main.yml:17]: Where/how is this used?  Looks like its used in the python role?  Is there a way to make that clearer?

File: tasks/install.yml [hunk 4]
```diff
@@ -88,19 +80,17 @@
...
+  known_hosts:
+    path: "{{ django_system_user_home }}/.ssh/known_hosts"
+    name: github.com
+    key: "{{ lookup('pipe', 'ssh-keyscan -t rsa github.com') }}"
   when:
-    - django_git_key is defined
-    - django_git_key is not none
-    - django_remove_git_key == True
+    - django_git_key_content is defined or django_git_key_ssh_file is defined
+    - django_git_key_content is not none or django_git_key_ssh_file is not none
```
Comment (DavisRayM) [inline tasks/install.yml:93]: Required while installing packages with pip using ssh `git+ssh`

**Comment by jasonrogena** [comment]
Looks good to me
